clientconn.go

```diff
@@ -960,8 +1020,22 @@ func (ac *addrConn) resetTransport() error {
 				Metadata:  addr.Metadata,
 				Authority: ac.cc.authority,
 			}
-			newTransport, err := transport.NewClientTransport(ac.cc.ctx, sinfo, copts, timeout)
+			connectCtx, cancel := context.WithDeadline(ac.ctx, connectDeadline)
+			done := make(chan struct{})
+			newTransport, err := transport.NewClientTransport(connectCtx, ac.cc.ctx, sinfo, copts, func() {
+				close(done)
+				ac.mu.Lock()
+				if !ac.backoffDeadline.IsZero() {
```

Comment:
It may be faster (and is simpler, code-wise) to just clear everything unconditionally.